avm/ptn/network/hub-networking/tests/e2e/defaults/main.test.bicep

```diff
@@ -0,0 +1,49 @@
+targetScope = 'subscription'
```

Comment:
I know this PR is still in draft, but I want to share some insights. 
There are a couple other contributors currently working on a pattern that already provided us with some feedback we're working on and may be of interest for you:
- We'll remove the requirement for the `defaults` & `waf-aligned` test for patterns. You are however invited to have them if you think they make sense for the pattern
- We added a feature that enables you to add post-deployment tests if you need them for your pattern to test something (which are also supported for res modules now). You can find an exampel [here](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/key-vault/vault/tests/e2e/max/tests/connectivity.tests.ps1). I'm calling this out here just because it was also a request by the other contributors ;)

Further, as we don't have a ptn module yet, I expect some issues here and there with the tooling. Hopefully there are none, but ... being first is always a gamble. IF you were to find any issues with e.g. the readme generation, tests, etc. please ping me / any of the other maintainers so that we can resolve them asap.

Reply:
@hundredacres regarding this point
> We'll remove the requirement for the defaults & waf-aligned test for patterns. You are however invited to have them if you think they make sense for the pattern

It is done now. You can go ahead merging latest main into the branch and retest the validation. Let us know if any additional blockers on the CI side.